animation_nodes/nodes/gpencil/replicate_gp_layer.py

```diff
@@ -0,0 +1,77 @@
+import bpy
+from bpy.props import *
+from mathutils import Matrix
+from ... data_structures import GPLayer, VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+transformationTypeItems = [
+    ("Matrix List", "Matrices", "", "NONE", 0),
+    ("Vector List", "Vectors", "", "NONE", 1)
+]
+
+class ReplicateGPLayerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ReplicateGPLayerNode"
+    bl_label = "Replicate GP Layer"
+
+    useLayerList: VectorizedSocket.newProperty()
+
+    transformationType: EnumProperty(name = "Transformation Type", default = "Matrix List",
+        items = transformationTypeItems, update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput(VectorizedSocket("GPLayer", "useLayerList",
+            ("Layer", "layer", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Layers", "layers")))
+        self.newInput(self.transformationType, "Transformations", "transformations")
+        self.newInput("Float List", "Offset Frame Numbers", "offsets", value = 0, hide = True)
+
+        self.newOutput("GPLayer List", "Layers", "outLayers")
+
+    def draw(self, layout):
+        layout.prop(self, "transformationType", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.transformationType == "Matrix List":
+            return "execute_MatrixList"
+        elif self.transformationType == "Vector List":
+            return "execute_VectorList"
+
+    def execute_MatrixList(self, layers, matrices, offsets):
+        if isinstance(layers, GPLayer):
+            layers = [layers]
+
+        _offsets = VirtualDoubleList.create(offsets, 0)
+
+        outLayers = []
+        for i, matrix in enumerate(matrices):
+            for layer in layers:
+                newLayer = layer.copy()
+                self.transformLayerByMatrix(newLayer, _offsets[i], matrix)
+                outLayers.append(newLayer)
+        return outLayers
+
+    def execute_VectorList(self, layers, vectors, offsets):
+        if isinstance(layers, GPLayer):
+            layers = [layers]
+
+        _offsets = VirtualDoubleList.create(offsets, 0)
+
+        outLayers = []
+        for i, vector in enumerate(vectors):
+            for layer in layers:
+                newLayer = layer.copy()
+                self.transformLayerByMatrix(newLayer, _offsets[i], vector)
```

Comment:
Should this be `transformLayerByVector`? Or just pass `Matrix.Translation(vector)` from here.

Reply:
Yes, I'll change it to `transformLayerByVector`.

Reply:
Or use `move` instead of `transform`.

Reply:
I have added the `move` for vectors.